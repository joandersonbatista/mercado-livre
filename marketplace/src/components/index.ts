export { Header } from './header/header';
export { IconButton } from './icon-button/icon-button';
export { Button } from './button/button';
export { Breadcrumbs } from './breadcrumbs/breadcrumbs';
export { BreadcrumbsSkeleton } from './breadcrumbs/breadcrumbs-skeleton';
export { Skeleton, type SkeletonProps } from './skeleton/skeleton';
export { LanguageSwitcher } from './language-switcher/language-switcher';
export { ServerError } from './server-error/server-error';
