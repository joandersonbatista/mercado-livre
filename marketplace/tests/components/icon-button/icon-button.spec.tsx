import { IconButton } from '@/components';
import { render, fireEvent } from '@testing-library/react';

describe('IconButton Component', () => {
  it('should render the button with the given icon', () => {
    const { getByTestId } = render(<IconButton icon={<span>🔍</span>} />);
    const buttonElement = getByTestId('icon-button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toContainHTML('🔍');
  });

  it('should apply the className correctly', () => {
    const { getByTestId } = render(
      <IconButton icon={<span>🔍</span>} className="custom-class" />,
    );
    const buttonElement = getByTestId('icon-button');

    expect(buttonElement).toHaveClass('icon-button');
    expect(buttonElement).toHaveClass('custom-class');
  });

  it('should forward additional props to the button element', () => {
    const { getByTestId } = render(<IconButton icon={<span>🔍</span>} disabled />);
    const buttonElement = getByTestId('icon-button');

    expect(buttonElement).toBeDisabled();
  });

  it('should match the snapshot', () => {
    const { container } = render(<IconButton icon={<span>🔍</span>} />);
    expect(container).toMatchSnapshot();
  });

  it('should call onClick when the button is clicked', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <IconButton icon={<span>🔍</span>} onClick={handleClick} />,
    );

    const buttonElement = getByTestId('icon-button');
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should not call onClick when the button is disabled', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <IconButton icon={<span>🔍</span>} onClick={handleClick} disabled />,
    );

    const buttonElement = getByTestId('icon-button');
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(0);
  });
});
