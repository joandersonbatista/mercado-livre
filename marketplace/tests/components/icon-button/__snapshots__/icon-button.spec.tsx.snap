// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component should match the snapshot 1`] = `
<div>
  <button
    class="icon-button "
    data-testid="icon-button"
  >
    <span>
      🔍
    </span>
  </button>
</div>
`;
